src/main/perl/Lock.pm

```diff
@@ -193,6 +235,61 @@ sub _try_lock
     return SUCCESS;
 }
 
+
+# Before CAF PR#132, CAF::Lock used the presence of the lock file as
+# condition to determine if the lock was taken or not.
+# The file would hold the PID of the process that had the lock
+# (and unlock simply removed the file).
+# This method provides support for the existence of an old-style lockfile
+# held by another process using old-style locking code.
+# It does not modify/cleanup the old-style file (even if it is stale).
+# Returns boolean true if the lock has been taken by another process
+# (i.e. if it is locked by another old-style instance).
+sub _is_locked_oldstyle
+{
+    my ($self, $force) = @_;
+
+    my $other_has_lock = 0;
+
+    # No file -> no lock
+    # Empty file (like a lock with current code) -> no old-style lock
+    my @st = stat($self->{LOCK_FILE});
```

Comment:
OK, that makes sense. Can we use the named attributes of stat instead of the numeric one? That would make the code more readable without having to grub around in man pages to remember what the $st[7] represents.

My other point was purely stylistic. The "common case" is that we are in a new-lock only situation. Therefore I should be able to read this check, see that it does nothing and ignore the rest of the code. 

```
if( rare-condition )  {
   # ugly stuff
}
return;
```
is fine until <<ugly stuff>> gets so long and deep that it's hard to figure out exactly where I return to the main code branch. In which case I think convention is to "short cut" the common case:
```
if (!rare-condition) {
  return;
} else {
  # long and deep code block
}
```
I think this is something recommended in the _Code Complete_ book but I don't remember anymore.

Thinking a bit more, an even better approach would be to move the stat() up to the calling point and hide the ugly stuff entirely in the subroutine.  

```
my @st = stat($self->{LOCK_FILE});
if (@st && $st->size > 0 && $self->_is_locked_oldstyle($force)) {
```